File: problem_010.pl
#!/usr/bin/env perl

print "Project Euler - Problem 10:
Find the sum of all the primes below two million.\n\n";

sub isPrime{
    local $i, $n;
    my($n) = @_;

    if ($n < 2){
        return 0;
    }

    if ($n == 2){
        return 1;
    }

    if ($n % 2 == 0){
        return 0;
    }

    for ($i=3 ; $i ** 2 <= $n ; $i+=2){
        if ($n % $i == 0){
            return 0;
        }
    }

    return 1;
}

$n = 3;
$s = 5;

while ($n < 2000000){
    $n += 2;
    if (isPrime($n) == 1){
        $s += $n;
    }
}

print $s, "\n";

File: problem_010.js
#!/usr/bin/node

console.log("Project Euler - Problem 10:");
console.log("Find the sum of all the primes below two million.\n");

const limit = 2000000;
var sum = 0;
var primes = [0,0];
for (var n=2; n<limit; n++)
    primes.push(n);

var p = 2;
while (p < limit){
    sum += p;

    // Remove all multiples of this prime
    let i = p + p;
    while (i < limit){
	primes[i] = 0;
	i += p;
    }

    // Find the next prime
    p += 1;
    while (primes[p] == 0)
	p += 1;
}

console.log(sum);
